vignettes/pull_request_comments.Rmd

```diff
@@ -0,0 +1,249 @@
+The `github_token` file should not have any file extensions.
+
+## Libraries
+
+To use the notebook, you will need to ensure you have the required R
+packages installed.
+
+```{r warning=FALSE,message=FALSE}
+rm(list = ls())
+require(kaiaulu)
+require(data.table)
+require(jsonlite)
+require(knitr)
+require(magrittr)
+require(gt)
+```
+
+## Project Configuration File
+
+To use the pipeline, you must specify the organization and project of
+interest, and your token.
+
+```{r warning=FALSE}
+conf <- parse_config("../conf/kaiaulu.yml")
+owner <- get_github_owner(conf, "project_key_1") # Has to match github organization (e.g. github.com/sailuh)
+repo <- get_github_repo(conf, "project_key_1") # Has to match github repository (e.g. github.com/sailuh/perceive)
+
+# Path you wish to save all raw data.
+save_path_issue_refresh <- get_github_issue_search_path(conf, "project_key_1")
+save_path_issue <- get_github_issue_path(conf, "project_key_1")
+save_path_pull_request <- get_github_pull_request_path(conf, "project_key_1")
+save_path_pr_comments <- get_github_pr_comments_path(conf, "project_key_1")
+save_path_issue_or_pr_comments <- get_github_issue_or_pr_comment_path(conf, "project_key_1")
+save_path_commit <- get_github_commit_path(conf, "project_key_1")
+save_path_pr_reviews <- get_github_pr_review_path(conf, "project_key_1")
+save_path_pr_commits <- get_github_pr_commits_path(conf, "project_key_1")
+save_path_pr_files <- get_github_pr_files_path(conf, "project_key_1")
+save_path_pr_reviewers <- get_github_pr_reviewers_path(conf, "project_key_1")
+save_path_pr_merge <- get_github_pr_merge_path(conf, "project_key_1")
+# Create all folder directories
+#create_file_directory(conf)
+
+# your file github_token (a text file) contains the GitHub token API
+token <- scan("~/.ssh/github_token",what="character",quiet=TRUE)
+```
+
+# GitHub Pull Request Comments
+
+## Initial Pull Request Comment
+
+The first pull request comment in the conversations tab is about the
+pull request itself. To get this, we must utilize the pull request
+endpoint to retrieve the body paragraph.
+
+```{r Collect information by pull request number}
+pull_number <- 330
+
+gh_response <- github_api_pull_request(owner, repo, pull_number, token)
+
+# create file name with path to save as json.
+file_name <- paste0(save_path_pull_request,
+                    owner,"_",repo,"_pullrequest","_", pull_number,
+                    ".json")
+#write to json.
+write_json(gh_response,file_name,pretty=TRUE,auto_unbox=TRUE)
+message("Written to file: ", file_name)
+```
+
+```{r}
+all_pull_requests <- lapply(list.files(save_path_pull_request,
+                                     full.names = TRUE),read_json)
+all_pull_requests <- lapply(all_pull_requests,
+                                   github_parse_pull_request)
+all_pull_requests <- rbindlist(all_pull_requests,fill=TRUE)
+
+head(all_pull_requests,1)  %>%
+  gt(auto_align = FALSE) 
+```
+
+## Issue Comments
```

Comment:
The section name is misleading. Are we not trying to get PR Comments here? Or is this about the first comment? Whichever the case, your section names should refer to PR only. Clarify on text if they came from issue.